api/authorizations.ts

```diff
@@ -1,144 +1,141 @@
 		try {
 			const method = req.query.m as string;
-			const token = getToken(req, res);
 			const request = req.query.req as string;
 
 			if (!request && method !== 'auth_chem') return res.status(404).json({ Message: "missing <req> string for request+authorisation number of the form req=AUTH_SST-AUTH_REQ" });
 			if (!req.query.date && method !== 'auth_chem') return res.status(404).json({ Message: "missing authorisation expiration <date>" });
 			const expirationDate = new Date(req.query.date as string);
 
-			let query = '';
-
 			switch (method) {
 				case "auth_req":
-					const idUnit = parseInt(req.query.id_unit as string);
-					if (!idUnit) return res.status(404).json({ Message: "missing <id_unit>" });
-
-					if (!req.query.room_ids) return res.status(404).json({ Message: "missing <room_ids> list of lab ids" });
-					const roomIds = (req.query.room_ids as string).split(',');
-
-					if (!req.query.scipers) return res.status(404).json({ Message: "missing <scipers> list of authorisation holders" });
-					const scipers = (req.query.scipers as string).split(',');
-
-					const cas = (req.query.cas as string).split(',');
-
-					query = `mutation addAuthorizationFromSNow {
-								addAuthorization(
-									token: "${token}",
-									id_unit: "${idUnit}",
-									authorization: "${request}",
-									creation_date: "${(new Date()).toLocaleDateString("en-GB")}",
-									expiration_date: "${(new Date(expirationDate)).toLocaleDateString("en-GB")}",
-									status: "Active",
-									type: "Chemical",
-									cas: [
-										${cas.map(c => `{name: "${c}", status: "New"}`)}
-									],
-									holders: [
-										${scipers.map(sc => `{sciper: ${sc}, status: "New"}`)}
-									],
-									rooms: [
-										${roomIds.map(r => `{id: ${r}, status: "New"}`)}
-									],
-								)
-							 {
-								errors {
-									message
-								}
-								isSuccess
-							}
-						}`;
-					const resultNew = await makeQuery(query, 'SNOW');
-					if (resultNew.addAuthorization.isSuccess)
-						res.json({Message: "Ok"});
+					if ( !req.user.canCallAPIToPostAuthorization )
+						res.status(403).json({Message: 'Unauthorized'});
 					else {
-						const error = resultNew.addAuthorization.errors.map(err => err.message).join(', ');
-						if (error.indexOf("Unique constraint failed on the constraint: `authorization`") > -1)
+						const idUnit = parseInt(req.query.id_unit as string);
+						if (!idUnit) return res.status(404).json({ Message: "missing <id_unit>" });
+
+						if (!req.query.room_ids) return res.status(404).json({ Message: "missing <room_ids> list of lab ids" });
+						const roomIds = (req.query.room_ids as string).split(',');
+
+						if (!req.query.scipers) return res.status(404).json({ Message: "missing <scipers> list of authorisation holders" });
+						const scipers = (req.query.scipers as string).split(',');
+
+						const cas = (req.query.cas as string).split(',');
```

Comment:
Please refactor that into a `validateCAS` imported function. It would probably be a great idea to add unit tests to it, while you are at it.